tools/bug-report/pkg/bugreport/bugreport.go

```diff
@@ -0,0 +1,302 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bugreport
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"istio.io/istio/operator/pkg/util"
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/tools/bug-report/pkg/archive"
+	cluster2 "istio.io/istio/tools/bug-report/pkg/cluster"
+	"istio.io/istio/tools/bug-report/pkg/config"
+	"istio.io/istio/tools/bug-report/pkg/content"
+	"istio.io/istio/tools/bug-report/pkg/filter"
+	"istio.io/istio/tools/bug-report/pkg/kubeclient"
+	"istio.io/istio/tools/bug-report/pkg/kubectlcmd"
+	"istio.io/istio/tools/bug-report/pkg/processlog"
+	"istio.io/pkg/log"
+	"istio.io/pkg/version"
+)
+
+const (
+	bugReportDefaultMaxSizeMb = 500
+	bugReportDefaultTimeout   = 30 * time.Minute
+)
+
+var (
+	bugReportDefaultIstioNamespace = "istio-system"
+	bugReportDefaultInclude        = []string{""}
+	bugReportDefaultExclude        = []string{"kube-system,kube-public"}
+)
+
+// Cmd returns a cobra command for bug-report.
+func Cmd() *cobra.Command {
+	rootCmd := &cobra.Command{
+		Use:          "bug-report",
+		Short:        "Cluster information and log capture support tool.",
+		SilenceUsage: true,
+		Long: "This command selectively captures cluster information and logs into an archive to help " +
+			"diagnose problems. It optionally uploads the archive to a GCS bucket.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runBugReportCommand(cmd)
+		},
+	}
+	rootCmd.AddCommand(version.CobraCommand())
+	addFlags(rootCmd, gConfig)
+
+	return rootCmd
+}
+
+var (
+	// Logs, along with stats and importance metrics. Key is path (namespace/deployment/pod/cluster) which can be
+	// parsed with ParsePath.
+	logs       = make(map[string]string)
+	stats      = make(map[string]*processlog.Stats)
+	importance = make(map[string]int)
+	// Aggregated errors for all fetch operations.
+	gErrors util.Errors
+	lock    = sync.RWMutex{}
+)
+
+func runBugReportCommand(_ *cobra.Command) error {
+	config, err := parseConfig()
+	if err != nil {
+		return err
+	}
+
+	clientConfig, clientset, err := kubeclient.New(config.KubeConfigPath, config.Context)
+	if err != nil {
+		return fmt.Errorf("could not initialize k8s client: %s ", err)
+	}
+	client, err := kube.NewExtendedClient(clientConfig, "")
```

Comment:
Replace with `kube.NewExtendedClient(kube.BuildClientCmd(kubeconfig, configContext)` and you can remove the `kubeclient.New` and associated code duplication

Reply:
we still need clientset but i'll take a look to see if this can be simplified per your suggestion.